certora/specs/SocialRecoveryModule.spec

```
@@ -207,3 +211,42 @@ rule revokeGuardianRevertPossibilities(env e, address otherAccount, address prev
         !safeContract.isModuleEnabled(currentContract) ||
         guardianStorageContract.entries[safeContract].guardians[prevGuardian] != otherAccount;
 }
+
+// This rule verifies that the guardian can always initiate recovery considering some ideal conditions.
+rule confirmRecoveryCanAlwaysBeInitiatedByGuardian(env e, address guardian, address[] newOwners, uint256 newThreshold, bool execute, uint256 index) {
+    require index < newOwners.length;
```

ubernit: Index isn't part of the rule parameterization, but rather used internally for verifying integrity. Just a small suggestion so it doesn't "leak" to the rule signature.

```suggestion
rule confirmRecoveryCanAlwaysBeInitiatedByGuardian(env e, address guardian, address[] newOwners, uint256 newThreshold, bool execute) {
    uint256 index;
    require index < newOwners.length;
```

---

Done 👍🏾